lib/models/hooks.js

```diff
@@ -0,0 +1,43 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = HooksModelFactory;
+
+HooksModelFactory.$provide = 'Models.Hooks';
+HooksModelFactory.$inject = [
+    'Model',
+    'Services.Configuration',
+    'uuid'
+];
+
+function HooksModelFactory (Model, configuration, uuid) {
+
+    return Model.extend({
+        connection: configuration.get('databaseType', 'mongo'),
+        identity: 'hooks',
+        attributes: {
+            id: {
+                type: 'string',
+                uuidv4: true,
+                primaryKey: true,
+                unique: true,
+                required: true,
+                defaultsTo: function() { return uuid.v4(); }
+            },
+            name: {
+                type: 'string',
```

Comment:
@pengz1 My criticism here was mainly around the upsert-like create method you had in your on-http PR which expected names to be unique.  Since you've changed this, I think the proposed semantics and implementation should be fine and are inline with the conventions we've used elsewhere.